src/darwin/Framework/CHIPTests/MTRCommissionableBrowserTests.m

```diff
@@ -0,0 +1,266 @@
+static const uint16_t kTestVendorId = 0xFFF1u;
+static const uint16_t kTestProductId = 0x8001u;
+static const uint16_t kTestDiscriminator = 3840u;
+static const uint16_t kDiscoverDeviceTimeoutInSeconds = 10;
+
+// Singleton controller we use.
+static MTRDeviceController * sController = nil;
+
+// Keys we can use to restart the controller.
+static MTRTestKeys * sTestKeys = nil;
+
+static BOOL sStackInitRan = NO;
+static BOOL sNeedsStackShutdown = YES;
+
+@interface DeviceScannerDelegate : NSObject <MTRCommissionableBrowserDelegate>
+@property (nonatomic) XCTestExpectation * expectation;
+
+- (instancetype)initWithExpectation:(XCTestExpectation *)expectation;
+- (void)didDiscoverCommissionable:(MTRDeviceController *)controller device:(MTRCommissionableBrowserResult *)device;
+- (void)commissionableUnavailable:(MTRDeviceController *)controller device:(MTRCommissionableBrowserResult *)device;
+@end
+
+@implementation DeviceScannerDelegate
+- (instancetype)initWithExpectation:(XCTestExpectation *)expectation
+{
+    if (!(self = [super init])) {
+        return nil;
+    }
+
+    _expectation = expectation;
+    return self;
+}
+
+- (void)didDiscoverCommissionable:(MTRDeviceController *)controller device:(MTRCommissionableBrowserResult *)device
+{
+    [self.expectation fulfill];
+
+    __auto_type instanceName = device.instanceName;
+    __auto_type vendorId = device.vendorID;
+    __auto_type productId = device.productID;
+    __auto_type discriminator = device.discriminator;
+    __auto_type commissioningMode = device.commissioningMode;
+
+    XCTAssertEqual([instanceName length], 16); // The  instance name is random, so just ensure the len is right.
+    XCTAssertEqualObjects(vendorId, @(kTestVendorId));
+    XCTAssertEqualObjects(productId, @(kTestProductId));
+    XCTAssertEqualObjects(discriminator, @(kTestDiscriminator));
+    XCTAssertEqual(commissioningMode, YES);
+
+    NSLog(@"Found Device (%@) with discriminator: %@ (vendor: %@, product: %@)", instanceName, discriminator, vendorId, productId);
+}
+
+- (void)commissionableUnavailable:(MTRDeviceController *)controller device:(MTRCommissionableBrowserResult *)device
+{
+    __auto_type instanceName = device.instanceName;
+    __auto_type vendorId = device.vendorID;
+    __auto_type productId = device.productID;
+    __auto_type discriminator = device.discriminator;
+
+    NSLog(
+        @"Removed Device (%@) with discriminator: %@ (vendor: %@, product: %@)", instanceName, discriminator, vendorId, productId);
+}
+@end
+
+@interface MTRCommissionableBrowserTests : XCTestCase
+
+@end
+
+@implementation MTRCommissionableBrowserTests
+
++ (void)tearDown
+{
+    // Global teardown, runs once
+    if (sNeedsStackShutdown) {
+        // We don't need to worry about ResetCommissionee.  If we get here,
+        // we're running only one of our test methods (using
+        // -only-testing:MatterTests/MTROTAProviderTests/testMethodName), since
+        // we did not run test999_TearDown.
```

Comment:
This comment is wrong, since there is no test999_TearDown in this test.  What is the actual model around commissioning here?

And also, we are clearly not running these tests with anything involving MTROTAProviderTests in the only-testing bit....